tests/backup/backup_delete_test.go

```diff
@@ -468,3 +469,226 @@ var _ = Describe("{DeleteIncrementalBackupsAndRecreateNew}", func() {
+				backupName := fmt.Sprintf("%s-%v", BackupNamePrefix, time.Now().Unix())
+				scheduleNames = append(scheduleNames, scheduleName)
+				labelSelectors := make(map[string]string)
+				err = CreateBackup(backupName, SourceClusterName, bkpLocationName, backupLocationUID, []string{namespace}, labelSelectors, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying backup %s creation", backupName))
+				backupNames = append(backupNames, backupName)
+			}
+		})
+
+		Step("Suspending the existing backup schedules", func() {
+			log.InfoD("Suspending the existing backup schedules")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, scheduleName := range scheduleNames {
+				scheduleUid, err := GetScheduleUID(scheduleName, orgID, ctx)
+				log.FailOnError(err, "Error while getting schedule uid %v", scheduleName)
+				err = suspendBackupSchedule(scheduleName, scheduleUid, periodicSchedulePolicyName, periodicSchedulePolicyUid, orgID, ctx)
+				dash.VerifySafely(err, nil, fmt.Sprintf("Verification of suspending backup schedule - %s", scheduleName))
+			}
+		})
+
+		Step("Delete the bucket where the backup objects are present", func() {
+			log.InfoD("Delete the bucket where the backup objects are present")
+			for _, provider := range providers {
+				globalBucketName := getGlobalBucketName(provider)
+				DeleteBucket(provider, globalBucketName)
+				log.Infof("Sleeping for default 10 minutes for next backup sync service to be triggered")
+				time.Sleep(10 * time.Minute)
+			}
+		})
+
+		Step("Verify the backups are in CloudBackupMissing state after bucket deletion", func() {
+			log.InfoD("Verify the backups are in CloudBackupMissing state after bucket deletion")
+			var wg sync.WaitGroup
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, backupName := range backupNames {
+				wg.Add(1)
+				go func(backupName string) {
+					defer GinkgoRecover()
+					defer wg.Done()
+					bkpUid, err := Inst().Backup.GetBackupUID(ctx, backupName, orgID)
+					log.FailOnError(err, "Fetching backup uid")
+					backupInspectRequest := &api.BackupInspectRequest{
+						Name:  backupName,
+						Uid:   bkpUid,
+						OrgId: orgID,
+					}
+					requiredStatus := api.BackupInfo_StatusInfo_CloudBackupMissing
+					backupCloudBackupMissingCheckFunc := func() (interface{}, bool, error) {
+						resp, err := Inst().Backup.InspectBackup(ctx, backupInspectRequest)
+						if err != nil {
+							return "", false, err
+						}
+						actual := resp.GetBackup().GetStatus().Status
+						if actual == requiredStatus {
+							return "", false, nil
+						}
+						return "", true, fmt.Errorf("backup status for [%s] expected was [%v] but got [%s]", backupName, requiredStatus, actual)
+					}
+					_, err = task.DoRetryWithTimeout(backupCloudBackupMissingCheckFunc, 20*time.Minute, 30*time.Second)
```

Comment:
Use `DoRetryWithTimeoutWithGinkgoRecover` (Mithun mentioned that all non-library functions must use for consistency)

Reply:
Done